### PR TITLE
[Feature](bangc-ops):voxelization support nan/inf.

### PR DESCRIPTION
**Pull request by mahxn0**
Thanks for your contribution and we appreciate it a lot. 

## 1. Motivation

support nan/inf case.

## 2. Modification
bangc-ops/kernels/voxelization/voxelization.cpp
bangc-ops/kernels/voxelization/voxelization_kernel.mlu

## 3. Test Report

If you want to know how to do operator testing, you can see [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).

#### 3.1.1 Accuracy Acceptance Standard

For static threshold standard details, see: [MLU-OPS Accuracy Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Accuracy-Acceptance-Standard.md).

- [ ] diff1: diff1 <= 3e-3
- [ ] diff2: diff2 <= 3e-3

#### 3.1.2 Operator Scheme checklist

|     No.        |                 Details              |            Check Results             |
|----------------|--------------------------------------|--------------------------------------|
|        1       |Supported hardware                    |             MLU370<br>MLU590         |
|        2       |Job types                             |          block <br> U1 <br> U4       |
|        3       |Layouts                               |          NHWC 、NCHW、ARRAY etc      |
|        4       |Whether multi-dimensions are supported|                                      |
|        5       |Whether element zero is supported     |                                      |
|        6       |Data type(half/float)                 |           half / float etc           |
|        7       |Whether there is size limit           |                                      |

#### 3.1.3 New Feature Test

If you have checked the following items, please tick the relevant box.

- [ ] Data type test
- [ ] Multi-dimensional tensor test
- [ ] Layout test
- [ ] Different size/integer remainder end segment/alignment misalignment test
- [ ] Zero dimensional tensor test/zero element test
- [x] stability test
- [x] Multiple platform test
- [x] Gen_case module test
- [x] Nan/INF tests 
- [x] Bug fix tests
- [ ] For memory leak check details, see[GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).
- [ ] For code coverage check details, see: [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).
- [ ] For I/O calculation efficiency check details, see: [MLU-OPS Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md).

#### 3.1.4 Parameter Check

When a new operator is submitted, the test points are given and the test results are stated.

|                   Test Point                    | Acceptance Standard | Test Result (Error Message) |
| ----------------------------------------------- | --------------------| --------------------------- |
| Whether it conforms to the operator restriction |     Normal error    |                             |
| Whether illegal parameters are passed           |     Normal error    |                             |

### 3.2 Accuracy Test

For the cases used in the New Feature Test section, the features and the number of cases are recorded here. When multiple operations are tested, multiple tables are needed to include details of these operations.

Operation:

|Test Point           | Description                      | Quantity |  Comment |
|----------           |----------------------------------|----------|  --------|
|Data type test       |half/float/int8                   |          |          |
|Mult-tensor test     |Supports 1-8 dims                 |          |          |
|Layout test          |Supports NCHW/NHWC                |          |          |
|Zero element test    |Whether to support this test      |          |          |
|Stability test       |--gtest_repeat=NUM<br>--thread=NUM|          |          |
|mlu-only mode test   |--mlu-only,see [MLU-OPS Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md)|          |          |
|Mult-platform test   |MLU370/MLU590                     |          |          |
|Nan/INF test         |Whether to support this test      |          |          |
|Memory leak check    |Test result                       |          |          |
|Code coverage check  |Test result                       |          |          |

### 3.3 Performance Test

See [MLU-OPS Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md) for details.

Platform：MLU370

|Operation|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-------|----|----|-----|----|----|----|-----|
|op_name|    |    |     |    |    |    |     |
|op_name|    |    |     |    |    |    |     |

Platform：MLU590

|Operation|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-------|----|----|----|----|----|----|-----|
|op_name|    |    |    |    |    |    |     |
|op_name|    |    |    |    |    |    |     |

### 3.4 Summary Analysis

Please give a brief overview here, if you want to note and summarize the content.

### CODE REVIEW

File: bangc-ops/kernels/voxelization/voxelization_kernel.mlu [hunk 1]
```diff
@@ -23,12 +23,21 @@
 #include "voxelization.h"
 
 #include <cmath>
-
 #include "kernels/kernel.h"
 #include "kernels/utils/common.h"
 
+#define INT_MIN -2147483648
```
Comment (defei-coder) [inline bangc-ops/kernels/voxelization/voxelization_kernel.mlu:29]: 使用 `#include <climits>` 中的 INT_MIN 宏更好些。直接使用这个数据可能会带来一些潜在的风险，下面是百度到的一些信息：在C/C++语言中，不能够直接使用-2147483648来代替最小负数，因为这不是一个数字，而是一个表达式。该表达式的意思是对整数21473648取负，但是2147483648已经溢出了int的上限，所以定义为（-INT_MAX -1）。

File: bangc-ops/kernels/voxelization/voxelization_kernel.mlu [hunk 2]
```diff
@@ -172,9 +181,24 @@ __mlu_global__ void mluDynamicVoxelize(const float *points,
...
+    grid_x = INT_MIN;
+  }
+  contain_nan_inf = containNanInf(coors_y_max, coors_y_min);
+  if (contain_nan_inf) {
+    grid_y = INT_MIN;
+  }
+  contain_nan_inf = containNanInf(coors_z_max, coors_z_min);
+  if (contain_nan_inf) {
+    grid_z = INT_MIN;
+  }
```
Comment (defei-coder) [inline bangc-ops/kernels/voxelization/voxelization_kernel.mlu:201]: 可以写成三目运算符的形式：
bool contain_nan_inf = false;
contain_nan_inf = containNanInf(coors_x_max, coors_x_min);
int32_t grid_x = contain_nan_inf ? INT_MIN : round((coors_x_max - coors_x_min) / voxel_x);

File: bangc-ops/kernels/voxelization/voxelization_kernel.mlu (hunk 1 above)
Comment (mahxn0) [reply bangc-ops/kernels/voxelization/voxelization_kernel.mlu]: 修改了

File: bangc-ops/kernels/voxelization/voxelization_kernel.mlu (hunk 2 above)
Comment (mahxn0) [reply bangc-ops/kernels/voxelization/voxelization_kernel.mlu]: 已修改

**Review by defei-coder — APPROVED** [review]

**Review by PetrelYy — APPROVED** [review]

**Review by DanieeelLiu — APPROVED** [review]

**Pull request merged**